ktx-library-codelab/step-03/myktxlibrary/src/main/java/com/example/android/myktxlibrary/ActivityUtils.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.myktxlibrary
+
+import android.Manifest
+import android.app.Activity
+import android.content.pm.PackageManager
+import android.location.Location
+import android.widget.TextView
+import androidx.core.app.ActivityCompat
+
+fun Activity.findAndSetText(id: Int, text: String) {
+    findViewById<TextView>(id).text = text
+}
+
+fun Activity.showLocation(id: Int, location: Location?) {
+    if (location != null) {
+        findAndSetText(id, location.asString(Location.FORMAT_MINUTES))
+    } else {
+        findAndSetText(id, "Location unknown")
+    }
+}
+
+fun Activity.hasPermission(permission: String): Boolean {
```

Comment:
Permission param is not used. Either remove the param or use it instead of hardcoding Manifest.permission.ACCESS_FINE_LOCATION

Reply:
this should be used of course! fixing